src/librustc_typeck/check/pat.rs

```diff
@@ -1390,9 +1393,8 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
 
     /// Type check the length of an array pattern.
     ///
-    /// Returns both the type of the variable length pattern
-    /// (or `tcx.err` in case there is none),
-    /// and the potentially inferred array type.
+    /// Returns both the type of the variable length pattern (or `None`), and the potentially
+    /// inferred array type. We should only return `None` for the slice type if `slice.is_none()`.
```

Comment:
```suggestion
    /// inferred array type. We only return `None` for the slice type if `slice.is_none()`.
```
It's a fact, not a wish. =P